runtime-k3d/src/deployment/build_vehicleapp.py

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) 2023 Robert Bosch GmbH
+#
+# This program and the accompanying materials are made available under the
+# terms of the Apache License, Version 2.0 which is available at
+# https://www.apache.org/licenses/LICENSE-2.0.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import subprocess
+from typing import List
+
+from yaspin import yaspin
+
+from velocitas_lib import get_app_manifest, get_workspace_dir, require_env
+
+
+def build_vehicleapp():
+    """Build VehicleApp docker image and display the progress using a spinner."""
+    with yaspin(text="Building VehicleApp...") as spinner:
+        try:
+            status = "> Building VehicleApp image"
+            app_name = get_app_manifest()["name"].lower()
+            image_tag = f"localhost:12345/{app_name}:local"
+            dockerfile_path = require_env("dockerfilePath")
+            os.environ["DOCKER_BUILDKIT"] = "1"
+            has_proxy: bool = os.getenv("HTTP_PROXY") is not None
+
+            extra_proxy_args: List[str] = list()
+
+            if has_proxy:
+                status = f"{status} with proxy configuration."
+                http_proxy = os.getenv("HTTP_PROXY", default="")
+                https_proxy = os.getenv("HTTPS_PROXY", default="")
+                ftp_proxy = os.getenv("FTP_PROXY", default="")
+                all_proxy = os.getenv("ALL_PROXY", default="")
+                no_proxy = os.getenv("NO_PROXY", default="")
+                extra_proxy_args = [
+                    "--build-arg",
+                    f"HTTP_PROXY={http_proxy}",
+                    "--build-arg",
+                    f"HTTPS_PROXY={https_proxy}",
+                    "--build-arg",
+                    f"FTP_PROXY={ftp_proxy}",
+                    "--build-arg",
+                    f"ALL_PROXY={all_proxy}",
+                    "--build-arg",
+                    f"NO_PROXY={no_proxy}",
+                ]
+            else:
+                status = f"{status} without proxy configuration."
```

Comment:
Could be simplified like this (just forwards those variable being set in the calling environment):
```suggestion
            extra_proxy_args: List[str] = [
                "--build-arg",
                "HTTP_PROXY",
                "--build-arg",
                "HTTPS_PROXY",
                "--build-arg",
                "FTP_PROXY",
                "--build-arg",
                "ALL_PROXY",
                "--build-arg",
                "NO_PROXY",
            ]
```

Reply:
did you test this behind proxy to be sure, i cannot since i dont have a proxy? I think the notation: f"HTTP_PROXY={http_proxy}" will be substituted with the value of the current environment variable. If you only want to pass the value from the environment automatically, this variable need to be part of the Dockerfile

Reply:
Tested - works and applied